# pa0 - Initial Entry

**Date:** 2025-06-22
**System:** Local
**Project:** pa-2025

## Objective

I am testing out using .md files instead of Benchling to track commands, and can then use github to keep track 
of these files.

This notebook (pa-2025) is for new PA analyses and is starting in June 2025. The purpose is to re-analyze PA 
transmission data and add some SNP analyses. The old analyses are not completely invalid, but my understanding 
of some steps has changed, and I want a fresh start. It will also let me mirror what I do for PA with SA and 
advance that project as well.

## Inputs

- There are **1427 "good" PA genomes sequenced**
- Trimmed reads: `/work/parkins_lab/project/conrad/pa-2025/fastq-files/trimmed-fastq/`
- Raw reads: `/work/parkins_lab/project/conrad/pa-2025/fastq-files/raw-fastq/`

Trimmed reads were generated as part of previous PA projects/analyses. They were generated by Trimmomatic 
(likely v0.39) with appropriate adapters (as determined by FastQC) removed. Other parameters used would have 
been: `ILLUMINACLIP:4:5`, `CROP:<appropriate length>`, and `MINLEN:<31 or 35>`.

Raw reads for most samples are in the above folder, but some samples seem to be missing. Raw reads for these 
samples are likely buried in other folders in `/work/parkins_lab/project/conrad/` or `/home/conrad.izydorczyk/`. 
Trimmed reads are the most important, so this is not critical at this point...

## Commands/Pipeline

## Output

## Notes

## Related Analyses
